discovery_engine_core/web-api/src/routes.rs

```diff
@@ -64,6 +68,53 @@ fn user_path() -> impl Filter<Extract = (UserId,), Error = Rejection> + Clone {
         })
 }
 
+// GET /documents/:document_id/properties
+fn get_document_properties(
+    state: Arc<AppState>,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    document_path()
+        .and(warp::path("properties"))
+        .and(warp::get())
+        .and(with_state(state))
+        .and_then(handlers::handle_get_document_properties)
+}
+
+// PUT /documents/:document_id/properties
+fn put_document_properties(
+    state: Arc<AppState>,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    document_path()
+        .and(warp::path("properties"))
+        .and(warp::put())
+        .and(warp::body::content_length_limit(1024))
+        .and(warp::body::json())
+        .and(with_state(state))
+        .and_then(handlers::handle_put_document_properties)
+}
+
+// DELETE /documents/:document_id/properties
+fn delete_document_properties(
+    state: Arc<AppState>,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    document_path()
+        .and(warp::path("properties"))
+        .and(warp::delete())
+        .and(with_state(state))
+        .and_then(handlers::handle_delete_document_properties)
+}
+
+// PATH /documents/:document_id
+fn document_path() -> impl Filter<Extract = (DocumentId,), Error = Rejection> + Clone {
+    warp::path("documents")
+        .and(warp::path::param::<String>())
+        .and_then(|document_id: String| async move {
+            urlencoding::decode(&document_id)
+                .map_err(Error::DocumentIdUtf8Conversion)
+                .and_then(DocumentId::new)
+                .map_err(warp::reject::custom)
+        })
```

Comment:
maybe we can add to this filter also "properties" path segment, instead of repeating it in every endopoint, wdyt?
```rust
.and(warp::path("properties"))
```